notation.go

```diff
@@ -262,3 +291,13 @@ func generateAnnotations(signerInfo *signature.SignerInfo) (map[string]string, e
 		annotationX509ChainThumbprint: string(val),
 	}, nil
 }
+
+func isDigestReference(reference string) bool {
```

Comment:
Instead doing your own way, you may try [Reference.ValidateReferenceAsDigest()](https://pkg.go.dev/oras.land/oras-go/v2@v2.0.0-rc.5/registry#Reference.ValidateReferenceAsDigest)